aio/tools/transforms/cli-docs-package/readers/cli-command.js

```
@@ -26,22 +26,59 @@ module.exports = function cliCommandFileReader(log) {
+            {text: name, path},
           ]
         });
+        if (doc.longDescription) {
+          doc.longDescriptionDoc = createLongDescriptionDoc(fileInfo);
+        }
         return [result];
       } catch (e) {
         log.warn(`Failed to read cli command file: "${fileInfo.relativePath}" - ${e.message}`);
       }
     }
   };
-};
+  function computeAliases(doc) {
+    return [doc.name].concat(doc.aliases || []).map(alias => `cli-${alias}`);
+  }
+
+  /**
+   * Synthesize a doc for the CLI command long description, which is used to generate links
+   * for viewing and editing the long description in GitHub.
+   *
+   * The long description is stored in a markdown file that is referenced from the original
+   * schema file for the command, via the `$longDescription` field. The field is a relative path
+   * to the markdown file from the schema file.
+   *
+   * This function tries to retrieve that original schema based on the file path of the help JSON
+   * file, which was passed to the `cliCommandFileReader.getDocs()` method.
+   */
+  function createLongDescriptionDoc(fileInfo) {
+    try {
+      const path = require('canonical-path');
+      const fs = require('fs');
+      const json5 = require('json5');
 
-function computeAliases(doc) {
-  return [doc.name].concat(doc.aliases || []).map(alias => `cli-${alias}`);
-}
\ No newline at end of file
+      const schemaJsonPath = path.resolve(fileInfo.basePath, '../commands', fileInfo.relativePath);
+      const schemaJson = fs.readFileSync(schemaJsonPath);
+      const schema = json5.parse(schemaJson);
+      if (schema.$longDescription) {
+        return {
+          docType: 'content',
+          startingLine: 0,
+          fileInfo: {
+            realProjectRelativePath:
+                path.join(path.dirname(fileInfo.realProjectRelativePath), schema.$longDescription)
+          }
+        };
+      }
+    } catch (e) {
+      log.warn('Unable to read CLI long description file info', e, fileInfo);
```

Why not fail hard here?

---

I decided that not having a link for the description would be better than failing the build altogether. But perhaps you are right that we should fail so that it picks up on a breaking change to the layout of the CLI project.

---

There is a precedent for this further up the file in this `FileReader` where we catch similar errors and just log them as warnings... Perhaps that should be a hard fail too?

---

The more hard fails the better :tada: